youtube_dl/extractor/ciscolive.py

```diff
@@ -0,0 +1,182 @@
+            "md5": "993d4cf051f6174059328b1dce8e94bd",
+            "info_dict": {
+                "id": "5803751616001",
+                "ext": "mp4",
+                "timestamp": 1530316421,
+                "title": "DevNet Panel-Applying Design Thinking to Building Products in Cisco",
+                "uploader_id": "5647924234001",
+                "upload_date": "20180629",
+            },
+        },
+        {
+            "url": "https://ciscolive.cisco.com/on-demand-library/?search.technology=scpsTechnology_applicationDevelopment&search.technology=scpsTechnology_ipv6&search.focus=scpsSessionFocus_troubleshootingTroubleshooting#/",
+            "md5": "80e0c3b87e373fe3a3316b934b8915bf",
+            "info_dict": {
+                "id": "5803735679001",
+                "ext": "mp4",
+                "timestamp": 1530311842,
+                "title": "Beating the CCIE Routing & Switching",
+                "uploader_id": "5647924234001",
+                "upload_date": "20180629",
+            },
+        },
+        {
+            "url": "https://ciscolive.cisco.com/on-demand-library/?search.solutions=scpsSolutions_cleanair#/",
+            "md5": "80e0c3b87e373fe3a3316b934b8915bf",
+            "info_dict": {
+                "id": "5803735679001",
+                "ext": "mp4",
+                "timestamp": 1530311842,
+                "title": "Beating the CCIE Routing & Switching",
+                "uploader_id": "5647924234001",
+                "upload_date": "20180629",
+            },
+        },
+    ]
+
+    # These appear to be constant across all Cisco Live presentations
+    # and are not tied to any user session or event
+    RAINFOCUS_API_URL = "https://events.rainfocus.com/api/%s"
+    RAINFOCUS_APIPROFILEID = "Na3vqYdAlJFSxhYTYQGuMbpafMqftalz"
+    RAINFOCUS_WIDGETID = "n6l4Lo05R8fiy3RpUBm447dZN8uNWoye"
+    BRIGHTCOVE_URL_TEMPLATE = "http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s"
+
+    def _parse_rf_item(self, rf_item):
+        """ Parses metadata and passes to Brightcove extractor
+
+        """
+        # Metadata parsed from Rainfocus API result
+        # Not all of which is appropriate to pass to Brightcove extractor
+        # but might be nice to print to output
+
+        event_name = rf_item.get("eventName")
+        # Full event name [Cisco Live EMEA 2016]
+        # rf_id = rf_item.get('eventId')
+        # Rainfocus ID [14382715417240cleu16]
+        cl_id = rf_item.get("abbreviation")
+        # Cisco Live ID - Shorthand session ID [BRKCRS-2501]
+        title = rf_item.get("title")
+        # Full session title [Campus QoS Design-Simplified]
+        description = rf_item.get("abstract")
+        # Description [This session will apply Cisco's QoS strategy for rich media...]
+        # TODO: Needs safe get() method
+        presenter_name = rf_item["participants"][0]["fullName"]
```

Comment:
The coding conventions don't explain how to safely extract nested results. Should I implement something like [this](https://stackoverflow.com/a/40675868), or is there a built-in helper I missed?

Reply:
I added `.get()` methods wherever appropriate. Since nested values include lists that don't have a native safe `.get()` method, left as-is for now. Advise further correction if needed.

Reply:
`utils.try_get`.